src/main/java/Duke.java

```diff
@@ -1,10 +1,225 @@
+import java.util.HashMap;
+import java.util.Scanner;
+
 public class Duke {
+    private static boolean isListening;
+    private static Task[] tasks = new Task[100];
+    private static int numTasks;
+
+
+    public static void setIsListening(boolean keepListening) {
+        Duke.isListening = keepListening;
+    }
+
+    public static void PrintBorder() {
+        System.out.println("────────────────────────────────────────────────────────────");
+    }
+
+    public static void Exit() {
+        System.out.print("Goodbye! Thank you for using MAX.\n");
+    }
+
+    public static String[] splitIntoSubcommands(String command) {
+        // " -" splits the string into substrings
+        // if the substring starts with two dashes '--' and is preceded by whitespace
+        // For example: deadline my_task --by -xyz sunday-monday
+        // Will output: ["deadline my_task", "by -xyz sunday-monday"]
+        // For now, this does not account for accidental or adversarial '--' input at the start of words
+        // but this is an unlikely edge case that can be dealt with later
+        return command.split(" --");
+    }
+
+    public static HashMap<String, String> getSubcommandMap(String[] commandList) {
+        // Each command can be processed into a subcommand and its corresponding text value
+        // For example: deadline work on CS2113 --by Sunday
+        // Should map to: <"deadline", "work on CS2113">, <by, Sunday>
+        HashMap<String, String> commandMap = new HashMap<>();
+        for (String cmdStr : commandList) {
+            String cmd = extractCommandFromSubcommand(cmdStr);
+            String text = extractTextFromSubcommand(cmdStr);
+            commandMap.put(cmd, text);
+        }
+        return commandMap;
+    }
+
+    public static String extractCommandFromSubcommand(String subcommand) {
+        // Subcommands have format <subcommand> <text_to_extract>
+        // Return the <subcommand>
+        // For example, the deadline subcommand: "by Friday, 13th"
+        // Should return "by"
+        String[] words = subcommand.split(" ");
+        return words[0];
+    }
+
+    public static String extractTextFromSubcommand(String subcommand) {
+        // Subcommands have format <subcommand> <text_to_extract>
+        // Return the <subcommand> and <text_to_extract> in a string array
+        // For example, the deadline subcommand: "by Friday, 13th"
+        // Should return ["by" , "Friday, 13th"]
+        String[] words = subcommand.split(" ");
+        String subcommandText = "";
+        for (int i = 1; i < words.length; ++i) {
+            subcommandText = subcommandText.concat(words[i]);
+            if (i != words.length - 1) {
+                subcommandText = subcommandText.concat(" ");
+            }
+        }
+        return subcommandText;
+    }
+
+    public static Task createTask(HashMap<String, String> commandMap, String taskType) {
```

Comment:
Hmm, its long due to the presence of guard clauses for error handling. Perhaps the error checking could be refactored out into a function -- but I think that this may cause issues down the road with complexity.

I'll experiment to see what works best!